include/libtorrent/sha1_hash.hpp

```diff
@@ -262,7 +263,10 @@ namespace libtorrent
 		{
 			return std::string(reinterpret_cast<char const*>(&m_number[0]), size());
 		}
-
+		std::string to_hex() const
```

Comment:
that simplification is that you use a function that returns a ``std::string`` rather than printing into a separate buffer first. It doesn't need to be a member function to do that